src/System.Text.Json/tests/JsonArrayTests.cs

```diff
@@ -0,0 +1,344 @@
+
+            jsonArray.Insert(0, int.MaxValue);
+            Assert.Equal(int.MaxValue, ((JsonNumber)jsonArray[0]).GetInt32());
+
+            jsonArray.Insert(0, long.MaxValue);
+            Assert.Equal(long.MaxValue, ((JsonNumber)jsonArray[0]).GetInt64());
+
+            jsonArray.Insert(0, 3.14f);
+            Assert.Equal(3.14f, ((JsonNumber)jsonArray[0]).GetSingle());
+
+            jsonArray.Insert(0, 3.14);
+            Assert.Equal(3.14, ((JsonNumber)jsonArray[0]).GetDouble());
+
+            jsonArray.Insert(0, sbyte.MaxValue);
+            Assert.Equal(sbyte.MaxValue, ((JsonNumber)jsonArray[0]).GetSByte());
+
+            jsonArray.Insert(0, ushort.MaxValue);
+            Assert.Equal(ushort.MaxValue, ((JsonNumber)jsonArray[0]).GetUInt16());
+
+            jsonArray.Insert(0, uint.MaxValue);
+            Assert.Equal(uint.MaxValue, ((JsonNumber)jsonArray[0]).GetUInt32());
+
+            jsonArray.Insert(0, ulong.MaxValue);
+            Assert.Equal(ulong.MaxValue, ((JsonNumber)jsonArray[0]).GetUInt64());
+
+            jsonArray.Insert(0, decimal.One);
+            Assert.Equal(decimal.One, ((JsonNumber)jsonArray[0]).GetDecimal());
+        }
+
+        [Fact]
+        public static void TestAddingJsonArray()
+        {
+            var preferences = new JsonObject()
+            {
+                { "colours", (JsonNode) new JsonArray{ "red", "green", "blue" } }
+            };
+
+            Assert.IsType<JsonArray>(preferences["colours"]);
+            var colours = preferences["colours"] as JsonArray;
+            Assert.Equal(3, colours.Count);
+
+            string[] expected = { "red", "green", "blue" };
+
+            for (int i = 0; i < colours.Count; i++)
+            {
+                Assert.IsType<JsonString>(colours[i]);
+                Assert.Equal(expected[i], colours[i] as JsonString);
+            }
+        }
+
+        [Fact]
+        public static void TestCretingJsonArrayFromStringArray()
+        {
+            string[] expected = { "sushi", "pasta", "cucumber soup" };
+            var preferences = new JsonObject()
+            {
+                { "dishes", (JsonNode) new JsonArray(expected) }
+            };
+
+            Assert.IsType<JsonArray>(preferences["dishes"]);
+            var dishesJson = preferences["dishes"] as JsonArray;
+            Assert.Equal(3, dishesJson.Count);
+
+            for (int i = 0; i < dishesJson.Count; i++)
+            {
+                Assert.IsType<JsonString>(dishesJson[i]);
+                Assert.Equal(expected[i], dishesJson[i] as JsonString);
+            }
+        }
+
+        /// <summary>
+        /// Adding JsonArray to JsonObject by passing JsonNumber array
+        /// </summary>
+        [Fact]
+        public static void TestAddingJsonArrayFromJsonNumberArray()
+        {
+            var preferences = new JsonObject()
+            {
+                { "prime numbers", new JsonNumber[] { 19, 37 } }
```

Comment:
This seems to be a JsonObject.Add(string, JsonNumbers) test, so it belongs in JsonObject tests.  (I'll stop pointing this out on this iteration)